src/NuGet.Clients/NuGet.PackageManagement.VisualStudio/PackageFeeds/InstalledAndTransitivePackageFeed.cs

```diff
@@ -63,5 +78,39 @@ internal override async Task<IPackageSearchMetadata> GetPackageMetadataAsync<T>(
 
             return await base.GetPackageMetadataAsync(identity, includePrerelease, cancellationToken);
         }
+
+        internal static PackageCollectionItem GetLatestApplicableTransitivePackageVersion(IGrouping<string, PackageCollectionItem> groupedPackagesById, IEnumerable<PackageCollectionItem> installedPkgs)
```

Comment:
It looks like this method is called in a Linq function, once for each group. Are there expected to be many groups for which this is called? If so, consider unrolling the Linq statements to reduce allocations for each of the calls if this is a hot code path. Otherwise, it should be fine if the number is always very small.